conf/app.routes

```diff
@@ -1,27 +1,28 @@
-GET         /api/digital-services/:name          @uk.gov.hmrc.teamsandrepositories.controller.LegacyRepositoriesController.digitalServiceDetails(name)
-GET         /api/digital-services                @uk.gov.hmrc.teamsandrepositories.controller.LegacyRepositoriesController.digitalServices
+GET         /api/repositories/:name                              @uk.gov.hmrc.teamsandrepositories.controller.LegacyRepositoriesController.repositoryDetails(name)
+GET         /api/repositories                                    @uk.gov.hmrc.teamsandrepositories.controller.LegacyRepositoriesController.repositories(archived: Option[Boolean])
+GET         /api/digital-services/:name                          @uk.gov.hmrc.teamsandrepositories.controller.LegacyRepositoriesController.digitalServiceDetails(name)
+GET         /api/digital-services                                @uk.gov.hmrc.teamsandrepositories.controller.LegacyRepositoriesController.digitalServices
 
-GET         /api/jenkins-url/:service            @uk.gov.hmrc.teamsandrepositories.controller.JenkinsController.lookup(service)
+GET         /api/jenkins-url/:service                            @uk.gov.hmrc.teamsandrepositories.controller.JenkinsController.lookup(service)
 
 # admin
-GET         /api/cache/reload                    uk.gov.hmrc.teamsandrepositories.controller.AdminController.reloadCache
-POST        /api/cache/clear                     uk.gov.hmrc.teamsandrepositories.controller.AdminController.clearCache
+GET         /api/cache/reload                                    uk.gov.hmrc.teamsandrepositories.controller.AdminController.reloadCache
+POST        /api/cache/clear                                     uk.gov.hmrc.teamsandrepositories.controller.AdminController.clearCache
 
 
 # v2 api
-GET         /api/v2/repositories                @uk.gov.hmrc.teamsandrepositories.controller.v2.TeamsAndRepositoriesController.allRepos(team: Option[String], archived: Option[Boolean], repoType: Option[RepoType])
-GET         /api/v2/repositories/:repoName      @uk.gov.hmrc.teamsandrepositories.controller.v2.TeamsAndRepositoriesController.findRepo(repoName)
-GET         /api/v2/teams                       @uk.gov.hmrc.teamsandrepositories.controller.v2.TeamsAndRepositoriesController.allTeams()
+GET         /api/v2/repositories                                 @uk.gov.hmrc.teamsandrepositories.controller.v2.TeamsAndRepositoriesController.allRepos(team: Option[String], archived: Option[Boolean], repoType: Option[RepoType])
+GET         /api/v2/repositories/:repoName                       @uk.gov.hmrc.teamsandrepositories.controller.v2.TeamsAndRepositoriesController.findRepo(repoName)
+POST        /api/v2/repositories/:repoName/set-branch-protection @uk.gov.hmrc.teamsandrepositories.controller.v2.TeamsAndRepositoriesController.setBranchProtection(repoName)
```

Comment:
This is the new route. I didn't put much thought into it though, so open to suggestions. Would we prefer a payload to a path param for the repo name?

Reply:
Maybe `/api/v2/repositories/:repoName/branch-protection/enable`?
This reserves the branch-protection as a resource type (e.g. could be read independently)

Or even `/api/v2/repositories/:repoName/branch-protection/enabled` with value `true` - this would be pointing at a property of branch-protection (i.e. noun `enabled` rather than verb `enable` - we alread have the verb `POST`)